src/Sentry/Protocol/Envelope.cs

```diff
@@ -97,7 +102,9 @@ public static Envelope FromEvent(SentryEvent @event)
                 lastByte = await stream.ReadByteAsync(cancellationToken).ConfigureAwait(false);
             }
 
-            return Json.DeserializeFromByteArray<Dictionary<string, object>>(buffer.ToArray());
+            return
+                Json.DeserializeFromByteArray<Dictionary<string, object>?>(buffer.ToArray()) ??
+                throw new InvalidOperationException("Envelope header is malformed.");
```

Comment:
Could we keep the operator on the left, I always get confused why the lines are stacked.
```suggestion
                Json.DeserializeFromByteArray<Dictionary<string, object>?>(buffer.ToArray())
                  ?? throw new InvalidOperationException("Envelope header is malformed.");
```

Reply:
![image](https://user-images.githubusercontent.com/1935960/97457803-c47d1600-1942-11eb-982a-b8adcc9670be.png)
How about this? Wink wink